crates/faucet/src/lib.rs

```diff
@@ -127,53 +130,51 @@ impl Faucet {
             .inspect_err(|err| {
                 error!(?err, "failed to sync state");
             })?;
+        client.ensure_genesis_in_place().await?;
 
-        info!("Fetching faucet state from node");
+        client.add_account(&account, false).await?;
+        client.set_setting(DEFAULT_ACCOUNT_ID_SETTING.to_owned(), account.id()).await?;
 
-        let issuance = match client.import_account_by_id(account.id()).await {
-            Ok(()) => {
-                // SAFETY: if import was successful, the account is tracked by the client
-                let record = client.get_account(account.id()).await?.unwrap();
-                info!(
-                    commitment = %record.account().commitment(),
-                    nonce = %record.account().nonce(),
-                    "Received faucet account state from the node",
-                );
-                record.account().get_token_issuance()?
-            },
-            Err(_) => match client.add_account(&account, false).await {
-                Ok(()) => {
-                    info!(
-                        commitment = %account.commitment(),
-                        nonce = %account.nonce(),
-                        "Loaded state from account file"
-                    );
-                    account.get_token_issuance()?
-                },
-                Err(ClientError::AccountAlreadyTracked(_)) => {
-                    // SAFETY: account is tracked, so its present in the db
-                    let record = client.get_account(account.id()).await?.unwrap();
-                    info!(
-                        commitment = %record.account().commitment(),
-                        nonce = %record.account().nonce(),
-                        "Loaded state from existing local client db"
-                    );
-                    record.account().get_token_issuance()?
-                },
-                Err(err) => anyhow::bail!("failed to add account from file: {err}"),
-            },
-        };
+        if deploy {
+            let mut faucet = Self::load(config).await?;
+            let empty_tx_request = TransactionRequestBuilder::new().build()?;
+            faucet.submit_transaction(empty_tx_request).await?;
+        }
 
-        client.ensure_genesis_in_place().await?;
+        Ok(())
+    }
 
-        let faucet = BasicFungibleFaucet::try_from(&account)?;
-        let tx_prover: Arc<dyn TransactionProver> = match remote_tx_prover_url {
+    /// Loads the faucet with the given config.
+    ///
+    /// The account used is the default account set in the store, that is set on `Faucet::init`.
+    #[instrument(target = COMPONENT, name = "faucet.load", fields(account_id), skip_all, err)]
+    pub async fn load(config: &FaucetConfig) -> anyhow::Result<Self> {
```

Comment:
With this new approach, importing the faucet from the network is not being done anymore, right? Should we bring that functionality back?